### DISCUSSION TITLE
LSP Hover Document Content Filtering

### DISCUSSION THREAD

**Discussion by karimlevallois** [LazyVim/LazyVim#2937] [Q&A]
Is there any way to remove the '-—moduleHash:-1430169424-' sections that are appearing at the bottom of my lsp hovers?

<img width="1680" alt="Screenshot 2024-04-05 at 18 54 05" src="https://github.com/LazyVim/LazyVim/assets/80129273/5f2b02f2-898d-47a5-abb9-7ddf9d904088">

<img width="1680" alt="Screenshot 2024-04-05 at 18 56 10" src="https://github.com/LazyVim/LazyVim/assets/80129273/87cc4418-e077-44f0-bd9a-56cf25fa568f">

I know I can add the following to my lsp.lua as a handler, but that config then takes priority and messes up my default settings for floating windows (removes padding, border, winblend etc. from noice).

```lua
["textDocument/hover"] = function(_, result, ctx, config)
    config = config or {}
    if result.contents then
        result.contents.value = result.contents.value:gsub("<!--.*", "")
    end
    config.border = "single"
    vim.lsp.handlers.hover(_, result, ctx, config)
end,
```

Is there any way to filter out this string without messing up my default settings? @folke any help greatly received, but no rush, whenever you have a few minutes after you are back from vacation.

**Comment by dpetka2001** [comment]
Unfortunately, I don't have knowledge about this, that's why I didn't answer when I saw your post (I try to answer the questions I know). Sorry. Your best bet would be to ping the maintainer and wait for an answer after he comes back from vacation.

**Reply by karimlevallois** [reply]
Thanks for coming back, I did initially ping him, but thought he probably will have enough on his plate so thought against it. I will ping him though now and let him come back when he has time.

**Comment by karimlevallois** [comment]
@folke Still stuck on this so if you could spare your knowledge I would really appreciate it.

**Reply by dpetka2001** [reply]
What LSP server is giving you this?

**Reply by karimlevallois** [reply]
Pylance

**Reply by dpetka2001** [reply]
What version of pylance?

**Reply by karimlevallois** [reply]
I will have to look, but it’s a pretty recent version.

**Reply by dpetka2001** [reply]
Try the following
```lua
return {
  {
    "folke/noice.nvim",
    opts = function(_, opts)
      opts.lsp = {
        hover = {
          -- silent = true,
        },
      }
    end,
    config = function(_, opts)
      require("noice").setup(opts)
      local Hover = require("noice.lsp.hover")
      local require = require("noice.util.lazy")

      local Format = require("noice.lsp.format")
      local Util = require("noice.util")
      local Docs = require("noice.lsp.docs")
      local Config = require("noice.config")
      function Hover.on_hover(_, result, ctx)
        if not (result and result.contents) then
          if Config.options.lsp.hover.silent ~= true then
            vim.notify("No information available")
          end
          return
        end

        local message = Docs.get("hover")
        if result.contents then
          result.contents.value = result.contents.value:gsub("<!--.*", "")
        end

        if not message:focus() then
          Format.format(message, result.contents, { ft = vim.bo[ctx.bufnr].filetype })
          if message:is_empty() then
            if Config.options.lsp.hover.silent ~= true then
              vim.notify("No information available")
            end
            return
          end
          Docs.show(message)
        end
      end

      Hover.on_hover = Util.protect(Hover.on_hover)
    end,
  },
}
```
and see if that works for you. Basically, we're overriding the `hover` function that `noice` uses for the lsp hover information. 

I just edited the noice `hover` function to include your snippet about removing `<---` from `results.contents.value`.
